### ISSUE TITLE
Archival and locking of a thread doesn't notify followers

### ISSUE THREAD

**Issue by MoshiKoi** [codidact/qpixel#1202]
Apparently, neither thread archival nor locking notify notify followers or participants. I tested this on the dev server, creating a thread, commenting on it with my alt, and then archiving and locking; the alt account didn't recieve any notifications.

**Comment by cellio** [comment]
When considering notifications, one question we should ask is: what can the notified person do in response to this notification?  For new activity, the person probably wants to go read it, edit suggestions need review, and for pings, a response has been requested.  But we don't notify for deleted comments or posts, for example.

Should we notify for a locked or archived thread?  Maybe for archiving, as that's a "resolved" signal.  Locking is usually a calming action (arguments getting out of control or the like), and notifications might stir things up again.

**Comment by trichoplax** [comment]
@ws909 The problem they are trying to solve there is that you cannot let someone know that their comment is ready to delete without also leaving a comment of your own, which they then have to let you know you can delete, in an endless cycle.

With thread archiving the comments do not need to be deleted, so the commenters do not need to notify each other. In some cases one of the commenters may wish to notify the other, but in that specific case they can simply add a comment before archiving. This way thread followers are only notified when the person archiving the thread explicitly chooses to notify them. Personally I would prefer this rather than being notified every time a thread I'm following is archived.

If some people expect that archiving will notify thread followers, and other people expect that it will not, maybe we could make it explicit to the person archiving the thread. There could be a dialogue box when you click "Archive" that has a checkbox saying "Notify thread followers" that defaults to not ticked. Ticking the box could provide a text input allowing a message to be entered. This way, thread followers either do not get notified, or in cases where the thread archiver wishes to notify them, there is a reminder that this is an option.

**Comment by ws909** [comment]
> he problem they are trying to solve there is that you cannot let someone know that their comment is ready to delete without also leaving a comment of your own, which they then have to let you know you can delete, in an endless cycle.

No, then you've misunderstood. As I explained in my answer (I'm Andreas), the reason I left a new comment, was to notify Karl not about his comment being ready to be deleted, but that the issue had been resolved.  There's no mention of any endless cycles anywhere there. The current best way to have those comments deleted, was to flag as "no longer needed". Both me and Karl come from SE, so that's the solution we're used to, and neither of us considered that comment threads present an alternative solution to this problem, which I agree seems better suited, although not without its own issues. That's what this GH issue comes from; resolving an issue, and marking it resolved by archiving the comment thread, and then notifying the users involved, that it's been resolved. Overall, this is more complicated than what seems necessary at first sight, but if Karl only wanted to point out a typo, he should've just suggested an edit to my post. He wasn't sure about it, and potentially misunderstood my post, which is why I wanted to notify him once it was resolved.

> With thread archiving, the comments do not need to be deleted, **so the commenters do not need to notify each other**

So, well, in this particular case, and likely more to come, it is/was needed.

**Comment by trichoplax** [comment]
I understand the specific case you describe in your answer. The case I was referring to as an endless cycle is the "no, *you* hang up" routine mentioned in the question. I understand that both cases need to be addressed.

The fact that there are multiple cases with different requirements is why I suggest that always notifying thread followers of thread archiving will mean unwanted notifications for some of the cases.

In the case where the only reason for commenting is to let another user know that they can delete their comment now, archiving removes the need to comment, and no notification is needed.

In the case where a user wants to let another user know that a query is resolved, they can add a comment to say so, immediately before archiving the thread. There is still no automated notification required.

I added the final paragraph of my previous comment in case there is a concern that some users may expect the act of archiving to notify thread followers.
